File: frontend/src/services/cinemaService.js
import apiClient from './apiClient'

export const layDanhSachRap = (khuVuc) => apiClient.get('/cinemas', { params: khuVuc ? { khuVuc } : {} }).then((phanHoi) => phanHoi.data)
export const layChiTietRap = (id) => apiClient.get(`/cinemas/${id}`).then((phanHoi) => phanHoi.data)
export const themRapMoi = (duLieu) => apiClient.post('/cinemas/admin', duLieu).then((phanHoi) => phanHoi.data)
export const capNhatRap = (id, duLieu) => apiClient.put(`/cinemas/admin/${id}`, duLieu).then((phanHoi) => phanHoi.data)
export const layDanhSachPhong = (maRap) => apiClient.get(`/cinemas/${maRap}/rooms`).then((phanHoi) => phanHoi.data)
export const themPhong = (maRap, duLieu) => apiClient.post(`/cinemas/${maRap}/rooms/admin`, duLieu).then((phanHoi) => phanHoi.data)
export const capNhatPhong = (maRap, maPhong, duLieu) => apiClient.put(`/cinemas/${maRap}/rooms/admin/${maPhong}`, duLieu).then((phanHoi) => phanHoi.data)
export const xoaPhong = (maRap, maPhong) => apiClient.delete(`/cinemas/${maRap}/rooms/admin/${maPhong}`)
export const laySoDoGhePhong = (maRap, maPhong) => apiClient.get(`/cinemas/${maRap}/rooms/${maPhong}/seats`).then((phanHoi) => phanHoi.data)
export const capNhatSoDoGhe = (maRap, maPhong, danhSachGhe) => apiClient.put(`/cinemas/${maRap}/rooms/admin/${maPhong}/seats`, danhSachGhe).then((phanHoi) => phanHoi.data)

export const taoMuoiPhongMacDinh = () => Array.from({ length: 10 }, (_, i) => {
  const so = String(i + 1).padStart(2, '0')
  return { maPhong: `P${so}`, tenPhong: `Phòng ${so}` }
})

File: src/main/java/com/cinema/booking/service/CinemaService.java
package com.cinema.booking.service;

import com.cinema.booking.document.Cinema;
import com.cinema.booking.document.Cinema.Room;
import com.cinema.booking.dto.CinemaDto;
import com.cinema.booking.dto.RoomDto;
import com.cinema.booking.dto.RoomSeatLayoutDto;
import com.cinema.booking.document.Cinema.Seat;
import java.util.List;

public interface CinemaService {
    List<String> layDanhSachKhuVuc();
    List<Cinema> layDanhSachRap(String khuVuc);
    Cinema layChiTietRap(String id);
    Cinema themRapMoi(CinemaDto dto);
    Cinema capNhatRap(String id, CinemaDto dto);
    List<Room> layDanhSachPhong(String maRap);
    Room themPhong(String maRap, RoomDto dto);
    Room capNhatPhong(String maRap, String maPhong, RoomDto dto);
    void xoaPhong(String maRap, String maPhong);
    RoomSeatLayoutDto laySoDoGhePhong(String maRap, String maPhong);
    RoomSeatLayoutDto capNhatSoDoGhePhong(String maRap, String maPhong, List<Seat> danhSachGhe);
}
